runtime/api/src/main/java/org/qbicc/runtime/CNative.java

```diff
@@ -815,12 +815,10 @@ public final boolean isNotPositive() {
          * Dereference the pointer, returning what the pointer points to. This operation
          * does not necessarily directly translate to a physical memory operation.
          *
-         * Pointers to structure types cannot be dereferenced, but members can be accessed as structPtr.deref().member
-         * which would be represented in C as structPtr->member.
          *
          * @return the pointed-to value
          */
-        public native T deref();
+        public static native <T extends object, P extends ptr<T>> T deref(ptr<T> obj);
```

Comment:
This should stay as an instance method. The intrinsic should still work identically as an instance intrinsic, except that the instance contains the pointer instead of the first argument.